cpp/src/toppra/algorithm.cpp

```diff
@@ -1,14 +1,27 @@
 #include <cstddef>
 #include <iostream>
 #include <toppra/algorithm.hpp>
+#ifdef BUILD_WITH_qpOASES
 #include <toppra/solver/qpOASES-wrapper.hpp>
+#endif
+#ifdef BUILD_WITH_GLPK
+#include <toppra/solver/glpk-wrapper.hpp>
+#endif
 #include "toppra/toppra.hpp"
 
 namespace toppra {
 
 PathParametrizationAlgorithm::PathParametrizationAlgorithm(
-    LinearConstraintPtrs constraints, const GeometricPath &path)
-    : m_constraints(std::move(constraints)), m_path(path){};
+    LinearConstraintPtrs constraints, const GeometricPathPtr &path)
+    : m_constraints(std::move(constraints)), m_path(path),
+#ifdef BUILD_WITH_qpOASES
+    m_solver (std::make_shared<solver::qpOASESWrapper>())
+#elif BUILD_WITH_GLPK
+    m_solver (std::make_shared<solver::qpOASESWrapper>())
```

Comment:
This looks like a Typo.

Btw, I think we should delay solver initialization in the derived class to avoid explicit dependencies in the base PathParametrizationAlgorithm class.

Reply:
Good catch !